profile/httprouter/handler.go

```diff
@@ -61,6 +63,52 @@ func HTTPAddNewCallsProfile(c *gin.Context) {
 	}
 }
 
+func HTTPExecuteConfigProfile(c *gin.Context) {
+
+	logger.HttpLog.Infoln("ExecuteConfigProfile API called")
+	//var prof profCtx.Profile
+
+	/*requestBody, err := c.GetRawData()
+	if err != nil {
+		logger.HttpLog.Errorf("Get Request Body error: %+v", err)
+		problemDetail := models.ProblemDetails{
+			Title:  "System failure",
+			Status: http.StatusInternalServerError,
+			Detail: err.Error(),
+			Cause:  "SYSTEM_FAILURE",
+		}
+		c.JSON(http.StatusInternalServerError, problemDetail)
+		return
+	}*/
+
+	var profileWaitGrp sync.WaitGroup
+	config := factory.AppConfig
+	// start profile and wait for it to finish (success or failure)
+	// Keep running gnbsim as long as profiles are not finished
+	for _, profileVal := range config.Configuration.Profiles {
+		if !profileVal.Enable {
+			continue
+		}
+		profileWaitGrp.Add(1)
+
+		profile.InitProfile(profileVal, profCtx.SummaryChan)
+
+		go func(profileCtx *profCtx.Profile) {
+			defer profileWaitGrp.Done()
+			profile.ExecuteProfile(profileCtx, profCtx.SummaryChan)
+		}(profileVal)
+
+		if config.Configuration.ExecInParallel == false {
+			profileWaitGrp.Wait()
+		}
+	}
+
+	if config.Configuration.ExecInParallel == true {
+		profileWaitGrp.Wait()
+	}
+
+	c.JSON(http.StatusOK, gin.H{})
```

Comment:
I realized yesterday that if profile takes lot of time then GUI waits for long time. Perhaps you respond to GUI immediately and then continue execution in background. If new API request comes while previous execution is still in progress then response with 200 again.

Reply:
we should send the result of the execution in the response. had done it in the gui branch. will merge to this. its better than sending empty response.